### DISCUSSION TITLE
json-pdf & pdf format

### DISCUSSION THREAD

**Discussion by ghost** [pdf-association/pdf-issues#243] [Q&A] [answered]
Hi everyone.

#### concept
I said recently, that was my last question. But something caught my attention today, there is a format called [json-pdf](https://github.com/yogthos/json-to-pdf) that generates a pdf file based on the json format. My question would be if there is any official pdf specification in json.

#### ``json-pdf.json``
```json
{"right-margin":10,
 "subject":"Some subject",
 "creator":"Jane Doe",
 "doc-header":["inspired by", "William Shakespeare"],
 "bottom-margin":25,
 "pages":true,
 "author":"John Doe",
 "header":"Page header text appears on each page",
 "left-margin":10,
 "title":"Test doc",
 "size":"a4",
 "letterhead":["A simple Letter head"],
 "orientation":"landscape",
 "font":{"size":11},
 "footer":
 "Page footer text appears on each page (includes page number)",
 "top-margin":20}
```

**code-description:** As we can see in the code there are metadata that will be generated in the final pdf file.

#### benefits
1. main advantage is that it is a text format and not binary. *which facilitates human reading.*
2. json format works on databases like mongodb, ravendb, couchdb - so it would be easy to keep the pdf file information, since if the pdf file has the json format it would be easy to manage, maintain, administer this text data. *this facilitates the adoption of pdf and json format for storage in databases.*
3. json format is widely used in apis and works mostly in current browsers like firefox, google chrome, brave, vivaldi etc. *this would make it even easier to adopt the json-pdf format.*
4. Some of the reasons I listed is for pdf to be even more commercially widespread.
5. This allows you to check which data, which PDF scripts are attached in the document. Which allows to increase security for users.

#### problems
1. maybe this pdf-json file format is not acceptable by the pdf community
2. Most pdf editors and viewers accept binary pdf format and not json-pdf

**Comment by petervwyatt** [comment, marked as answer]
There is no standardized JSON-based PDF "dumps" as all the tools I am aware of define their JSON schemas quite differently. They also all make various assumptions in their conversion and in their ability to support all PDF features which means that you cannot always rely on the JSON - it really depends on what the intent of your forensic analysis or debugging is trying to achieve.

History lesson: back in 2006-2007 (before JSON was so popular) there was a research project by Adobe Labs called Mars which expressed PDF purely in XML. Look for a paper called "The Mars Project — PDF in XML" by Matthew Hardy. Adobe Labs also published a full spec for Mars and a plugin for Acrobat of the day that could read and write Mars XML files. This never gained traction even though it was the "preferred" human readable format of the era...

**Reply by ghost** [reply]
Hi petervwyatt, thank you for feedback.
